containers/clojure/.devcontainer/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# [Choice] Java version (use -bullseye variants on local arm64/Apple Silicon): 11, 16, 11-bullseye, 16-bullseye, 11-buster, 16-buster
+ARG VARIANT=11-bullseye
+FROM mcr.microsoft.com/vscode/devcontainers/java:${VARIANT}
+
+# [Option] Clojure version
```

Comment:
The `[Option]` and `[Choice]` comments result in something appearing in the "Add Dev Container Configuration" UX when this definition is selection.  An `[Option]` will set the argument that follows it to a boolean true/false while a `[Choice]` will allow you to pick from the comma separated list that follows it the main comment.

Right now, you'd be prompted to pick a java version from the list, but a `CLOJURE_VERSION` would result be a true or false value - which I'm assuming wouldn't work.  The other question is how many of these options you'd like to have show up in that UX. You can keep them as options/choices that don't show up on the UX by just omitting the `[Option]` or `[Choice]` part of the comment, or if you'd perfer, use `[Optional]` since this would be ignored.

If you'd prefer not to have these appear in the "Add Dev Container Configuration Files" UX, you can use "Optional" instead of option, or just remove the `[Option]` part of the comment so people can make these settings manually.

Reply:
Okay, I have moved these to "Optional" to suppress display in the UX. My thinking is that by using "Optional", it might hint to people that these are values they should be able to change without fear of breaking the container.

I have also updated the list of Java variants to match the list in the Java devcontainer.